ui/admin-portal/src/app/components/chat/chat-read-status/chat-read-status.component.ts

```diff
@@ -0,0 +1,85 @@
+import {Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {combineLatest, Subscription} from "rxjs";
+import {ChatService} from "../../../services/chat.service";
+import {JobChat} from "../../../model/chat";
+import {map} from "rxjs/operators";
+
+/**
+ * Component which takes an array of chats and sets the unreadIndicator to unread ('*") if
+ * any of the chats is unread.
+ * Otherwise it sets the unread indicator to blank.
+ */
+@Component({
+  selector: 'app-chat-read-status',
+  templateUrl: './chat-read-status.component.html',
+  styleUrls: ['./chat-read-status.component.scss']
+})
+export class ChatReadStatusComponent implements OnInit, OnChanges, OnDestroy {
+
+  @Input() chats: JobChat[];
+
+  unreadIndicator: string;
+
+  private subscription: Subscription;
+
+  constructor(
+    private chatService: ChatService
+  ) { }
+
+  ngOnInit(): void {
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (this.chats && this.chats.length > 0) {
+      let allChatsRead =
+        this.chats.find(chat => !this.chatService.isChatRead(chat)) == null;
+      this.setIndicator(allChatsRead);
+      this.subscribeForChatUpdates();
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribe();
+  }
+
+  private subscribeForChatUpdates() {
+    if (this.chats) {
+      this.unsubscribe();
```

Comment:
```suggestion
      this.unsubscribe();
      this.unreadIndicator = ''
```

Will handle the case where there are no chats yet